oci/oci_linux.go

```diff
@@ -0,0 +1,57 @@
+// +build linux
+
+package oci
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/containerd/cgroups"
+	"github.com/kubernetes-incubator/cri-o/utils"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+func (r *Runtime) createContainerPlatform(c *Container, cgroupParent string, pid int) error {
+	// Move conmon to specified cgroup
+	if r.cgroupManager == SystemdCgroupsManager {
+		logrus.Infof("Running conmon under slice %s and unitName %s", cgroupParent, createUnitName("crio-conmon", c.id))
+		if err := utils.RunUnderSystemdScope(pid, cgroupParent, createUnitName("crio-conmon", c.id)); err != nil {
+			logrus.Warnf("Failed to add conmon to systemd sandbox cgroup: %v", err)
```

Comment:
I think we want `return fmt.Errorf("failed to add conmon to systemd sandbox cgroup: %v", err)` or even just `return err` here.  Decisions about logging and/or dying can be made by the `createContainerPlatform` caller.

Reply:
again, this is just carrying over existing behaviour, not introducing changes

Reply:
> again, this is just carrying over existing behaviour, not introducing changes

`createContainerPlatform` is new to this PR, so I think we want to decide whether *it* considers errors fatal or not.  I arge for "consider errors fatal" here.  When `CreateContainer` calls this function, *it* can decide those errors are non-fatal and turn them into log warnings ([for now][1]).  That preserves the existing overall behavior (logged warnings), while making it easier to tighten down later.

[1]: https://github.com/kubernetes-incubator/cri-o/pull/1407/files#r179118144

Reply:
please review the code this is copied from. I do not intend to change the behaviour in this PR. Please.

Reply:
> I do not intend to change the behaviour in this PR.
Returning errors here doesn't change the behavior, because you have the caller converting those errors into logged warnings.  But if you're really committed to making `createContainerPlatform` itself nonfatal, I don't see why you'd define it to return `error` at all.